### PR TITLE
*: do not set null column value to save storage space.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
This should be merged after TiKV coprocessor implements this feature.

@ngaut @shenli @BusyJay PTAL

File: table/tables/tables.go [hunk 1]
```diff
@@ -325,6 +325,10 @@ func (t *Table) AddRecord(ctx context.Context, r []types.Datum) (recordID int64,
 		if col.IsPKHandleColumn(t.meta) {
 			continue
 		}
+		if col.DefaultValue == nil && r[col.Offset].Kind() == types.KindNull {
```
Comment (shenli) [inline table/tables/tables.go]: Check NotNil flag? If default value is not specified, col.DefaultValue will be nil.

Comment (shenli) [inline table/tables/tables.go]: Ignore this.

**Comment by shenli** [comment]
LGTM

**Comment by disksing** [comment]
I guess we have to update DDL reorg logic. For instance, when backfill columns for a "add column" DDL, we have to check if the column is null.

**Comment by coocood** [comment]
@disksing set the column value to null instead of do not set has the same effect.
But we can ignore reorg phase if the column default value is null in future PR.

**Comment by shenli** [comment]
@zimulala PTAL

File: executor/executor_test.go [hunk 2]
```diff
@@ -1165,3 +1165,19 @@ func (s *testSuite) TestInSubquery(c *C) {
 	tk.MustExec("insert t1 values (281.37)")
 	tk.MustQuery("select a from t1 where (a in (select a from t1))").Check(testkit.Rows("281.37"))
 }
+
+func (s *testSuite) TestDefaultNull(c *C) {
```
Comment (zimulala) [inline executor/executor_test.go]: Add `defer testleak.AfterTest(c)()` ?

**Comment by zimulala** [comment]
Rest LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**